tools/istio-iptables/pkg/builder/iptables_builder_impl.go

```diff
@@ -170,6 +177,120 @@ func (rb *IptablesRuleBuilder) buildRules(rules []*Rule) [][]string {
 	return output
 }
 
+func reverseRules(rules []*Rule) []*Rule {
+	output := make([]*Rule, 0)
+	for _, r := range rules {
+		var modifiedParams []string
+		skip := false
+		insertIndex := -1
+		for i, element := range r.params {
+			if insertIndex >= 0 && i == insertIndex+2 {
+				continue
+			}
+			if element == "-A" || element == "--append" {
+				modifiedParams = append(modifiedParams, "-D")
+			} else if element == "-I" || element == "--insert" {
+				insertIndex = i
+				modifiedParams = append(modifiedParams, "-D")
+			} else {
+				modifiedParams = append(modifiedParams, element)
+			}
+
+			if ((element == "-A" || element == "--append") || (element == "-I" || element == "--insert")) &&
+				i < len(r.params)-1 && strings.HasPrefix(r.params[i+1], "ISTIO_") {
+				skip = true
+			} else if (element == "-j" || element == "--jump") && i < len(r.params)-1 && strings.HasPrefix(r.params[i+1], "ISTIO_") {
+				skip = false // Override previous skip if this is a jump-rule
+			}
+		}
+		if skip {
+			continue
+		}
+
+		output = append(output, &Rule{
+			chain:  r.chain,
+			table:  r.table,
+			params: modifiedParams,
+		})
+	}
+	return output
```

Comment:
For any func (like this and `checkRules` and `cleanupRules`) that mutates the rules, can you debug-log-dump the result?

Reply:
Added debug dumps and also some additional debug messages for the verification logic that determines whether a delta is found and reconciliation is needed